tools/internal_ci/linux/grpc_xds_k8s_run_test.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## xDS test server/client Docker images
+readonly IMAGE_REPO="gcr.io/grpc-testing/xds-interop"
+
+find_latest() {
+  gcloud container images list-tags --filter=tags\~v1\.\\d\\d\.x "${IMAGE_REPO}/${1}-${2}" '--format=table[no-heading](tags)' | awk -F, '{print $2}' | sort | tail -n 1
+}
+
+if [ "${LATEST_BRANCH}" == "" ]; then
+  cpp_server=$(find_latest cpp server)
+  cpp_client=$(find_latest cpp client)
+  go_server=$(find_latest go server)
+  go_client=$(find_latest go client)
+  java_server=$(find_latest java server)
+  java_client=$(find_latest java client)
+  LATEST_BRANCH="$((echo $cpp_server; echo $cpp_client; echo $go_server; echo $go_client; echo $java_server; echo $java_client;) | sort -g | head -1)"
```

Comment:
Syntax highlighting is getting broken here. Could you please fix it? While not an issue on its own, this often indicates an underlying problem.

Reply:
You can use `shellcheck` command for linting:

```sh
In grpc_xds_k8s_run_test.sh line 30:
  LATEST_BRANCH="$((echo $cpp_server; echo $cpp_client; echo $go_server; echo $go_client; echo $java_server; echo $java_client;) | sort -g | head -1)"
                 ^-- SC1102 (error): Shells disambiguate $(( differently or not at all. For $(command substitution), add space after $( . For $((arithmetics)), fix parsing errors.
                         ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                           ^---------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                             ^--------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                                              ^--------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                                                               ^----------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                                                                                                  ^----------^ SC2086 (info): Double quote to prevent globbing and word splitting.
```

Reply:
Also - you don't need separate echo commands. You use a single statement for it, f.e.:

```sh
❯ echo "${USER}${SHELL}"
sergiitk/bin/zsh
```

Reply:
`echo "${USER}\n${SHELL}"` is not guaranteed to work. But I think `echo -e "${USER}\n${SHELL}"` would work. Some of these shell idiosyncrasies are weird and that's why I went with the most basic way that's not dependent on such quirks.

Reply:
done

Reply:
done